test/fs/test_idbfs_autopersist.c

```diff
@@ -0,0 +1,237 @@
+      assert(fd != -1);
+      break;
+    }
+    case 2: {
+      struct stat st;
+      int res = lstat("/working1/file", &st);
+      assert(res == 0);
+      assert(st.st_size == 0);
+      break;
+    }
+    default:
+      assert(false);
+  }
+}
+
+static void test_case_close() {
+  switch (TEST_PHASE) {
+    case 1: {
+      int fd = open("/working1/file", O_RDWR | O_CREAT | O_EXCL, 0777);
+      assert(fd != -1);
+      break;
+    }
+    case 2: {
+      int fd = open("/working1/file", O_RDWR | O_CREAT, 0777);
+      assert(fd != -1);
+      ssize_t bytes_written = write(fd, "foo", 3);
+      assert(bytes_written == 3);
+      int res = close(fd);
+      assert(res == 0);
+      break;
+    }
+    case 3: {
+      struct stat st;
+      int res = lstat("/working1/file", &st);
+      assert(res == 0);
+      assert(st.st_size == 3);
+      break;
+    }
+    default:
+      assert(false);
+  }
+}
+
+static void test_case_symlink() {
+  switch (TEST_PHASE) {
+    case 1: {
+      int fd = open("/working1/file", O_RDWR | O_CREAT | O_EXCL, 0777);
+      assert(fd != -1);
+      break;
+    }
+    case 2: {
+      int res = symlink("/working1/file", "/working1/symlink");
+      assert(res == 0);
+      break;
+    }
+    case 3: {
+      struct stat st;
+      int res = lstat("/working1/symlink", &st);
+      assert(res == 0);
+      break;
+    }
+    default:
+      assert(false);
+  }
+}
+
+static void test_case_unlink() {
+  switch (TEST_PHASE) {
+    case 1: {
+      int fd = open("/working1/file", O_RDWR | O_CREAT | O_EXCL, 0777);
+      assert(fd != -1);
+      break;
+    }
+    case 2: {
+      int res = unlink("/working1/file");
+      assert(res == 0);
+      break;
+    }
+    case 3: {
+      struct stat st;
+      int res = lstat("/working1/file", &st);
+      assert(res == -1);
+      assert(errno == ENOENT);
+      break;
+    }
+    default:
+      assert(false);
+  }
+}
+
+static void test_case_rename() {
+  switch (TEST_PHASE) {
+    case 1: {
+      int fd = open("/working1/file", O_RDWR | O_CREAT | O_EXCL, 0777);
+      assert(fd != -1);
+      break;
+    }
+    case 2: {
+      int res = rename("/working1/file", "/working1/file_renamed");
+      assert(res == 0);
+      break;
+    }
+    case 3: {
+      struct stat st;
+      int res = lstat("/working1/file_renamed", &st);
+      assert(res == 0);
+      res = lstat("/working1/file", &st);
+      assert(res == -1);
+      assert(errno == ENOENT);
+      break;
+    }
+    default:
+      assert(false);
+  }
+}
+
+static void test_case_mkdir() {
+  switch (TEST_PHASE) {
+    case 1: {
+      int res = mkdir("/working1/dir", 0777);
+      assert(res == 0);
+      break;
+    }
+    case 2: {
+      struct stat st;
+      int res = lstat("/working1/dir", &st);
+      assert(res == 0);
+      break;
+    }
+    default:
+      assert(false);
+  }
+}
+
+void finish() {
+  emscripten_force_exit(0);
+}
+
+void test() {
```

Comment:
You can mark `finish` and `test` as `EMSCRIPTEN_KEEPALIVE` and avoid passing `-sEXPORTED_FUNCTIONS` on the command line.

Reply:
Feel free to update that test too.. otherwise we can do it as a followup.